lib/css/components/link.less

```diff
@@ -68,6 +68,13 @@ a,
         }
     }
 
+    // MODIFIERS
+    fieldset[disabled] & {
```

Comment:
Move fieldset stuff all into `fieldset.less`. Two ways to think about it:

1. Styles are targeting the child, put them in the child file
2. Styles are in the context of the parent, put them in the parent file

This is not a hill I'm willing to die on, just food for thought.

Reply:
This was originally applied to `a,.s-btn,.s-link`. We're covering `a, .s-link`, but not `.s-btn` I believe.

https://github.com/StackExchange/Stacks/blob/develop/lib/css/components/inputs.less#L91-L93

Reply:
> Styles are in the context of the parent, put them in the parent file

I opted for option 2 in the spirit of component styles being self-contained. Basically, my goal is that anyone trying to see what could be affecting a given component's styling can look in a single file. Big picture, if we every decided to get rid of or rename a component, all the styles would (optimally) live in the component's `.less` file and hopefully be slightly easier to handle without leaving crust around our codebase.

---

> This was originally applied to a,.s-btn,.s-link. We're covering a, .s-link, but not .s-btn I believe.

I'm not sure how we missed it, but I added this style to [buttons.less](https://github.com/StackExchange/Stacks/blob/43e3ebe364639827fc12b7b0a525c388c56f1d77/lib/css/components/buttons.less#L60) in https://github.com/StackExchange/Stacks/commit/939f1025b0e3a4b108637f3cb9e1adf3c6b2f5ca